webrtc-kmp/build.gradle.kts

```diff
@@ -53,9 +53,49 @@ android {
 dependencies {
     commonMainImplementation(deps.kotlin.coroutines)
     androidMainImplementation(deps.androidx.coreKtx)
-    androidMainApi(deps.webrtcSdk)
+    androidMainApi(deps.webrtc.android)
     androidTestImplementation(deps.androidx.test.core)
     androidTestImplementation(deps.androidx.test.runner)
+
+    jvmMainApi(deps.webrtc.java)
+    jvmMainImplementation(
+        group = deps.webrtc.java.get().group!!,
+        name = deps.webrtc.java.get().name,
+        version = deps.webrtc.java.get().version,
+        classifier = "windows-x86_64"
+    )
+    jvmMainImplementation(
+        group = deps.webrtc.java.get().group!!,
+        name = deps.webrtc.java.get().name,
+        version = deps.webrtc.java.get().version,
+        classifier = "macos-x86_64"
+    )
```

Comment:
I've got on my Mac M2

```
java.lang.UnsatisfiedLinkError: /private/var/folders/sp/8h47vr6s72v4hngsrxrfwpzm0000gn/T/libwebrtc-java1779612920915696582.dylib: dlopen(/private/var/folders/sp/8h47vr6s72v4hngsrxrfwpzm0000gn/T/libwebrtc-java1779612920915696582.dylib, 0x0001): tried: '/private/var/folders/sp/8h47vr6s72v4hngsrxrfwpzm0000gn/T/libwebrtc-java1779612920915696582.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/sp/8h47vr6s72v4hngsrxrfwpzm0000gn/T/libwebrtc-java1779612920915696582.dylib' (no such file), '/private/var/folders/sp/8h47vr6s72v4hngsrxrfwpzm0000gn/T/libwebrtc-java1779612920915696582.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64'))
```

But it's okay if I comment this declaration. I'll try to figure out how to solve this if you don't have M1/M2 at the moment.

Reply:
I only have a M1 Mac Mini, so no camera or microphone to test. But also got it at least running, by, like you mentioned, removing the x86 dependency.

I think adding this native dependencies would need to be done by the implementing application. It still worked after I removed the import from the library build.gradle and added it to the sample app build.gradle.

I will remove the import from the library and update the README to reflect that it is neccessary for implementing applications to add the right dependency on their own (like you need to add the WebRTC pod to iOS apps right now).

Reply:
I think we have to leave import with classifier in order to run JVM tests in CI. I guess we could solve it as following:

```kotlin
    val osName = System.getProperty("os.name").lowercase()
    val hostOS = if (osName.contains("mac")) {
        "macos"
    } else if (osName.contains("linux")) {
        "linux"
    } else if (osName.contains("windows")) {
        "windows"
    } else {
        throw IllegalStateException("Unsupported OS: $osName")
    }
    val hostArch = System.getProperty("os.arch").lowercase()
    jvmMainApi(deps.webrtc.java)
    jvmMainImplementation(
        group = deps.webrtc.java.get().group!!,
        name = deps.webrtc.java.get().name,
        version = deps.webrtc.java.get().version,
        classifier = "$hostOS-$hostArch"
    )
```

It works well on my Mac. However, I'd like to test it on Windows/Linux too. What do you think?

Reply:
For the CI tests in this repo? Then as `jvmTestImplementation`? I am not that familiar with publishing artifacts and am not sure what the published artifact with one of them defined as a dependency would look like. 
But maybe the `webrtc-kmp` jvm artifact could also be published with every classifier and dependency separately instead?

Reply:
I had to add a switch for the architecture, because "os.arch" returned "amd64" on my windows device.
```
val hostArch = when(val arch = System.getProperty("os.arch").lowercase()) {
      "amd64" -> "x86_64"
      else -> arch
  }
```